rules/ast/functions.go

```diff
@@ -498,6 +499,43 @@ func derivImpl(timestamp clientmodel.Timestamp, args []Node) interface{} {
 	return resultVector
 }
 
+// === histogram_quantile(k ScalarNode, vector VectorNode) Vector ===
```

Comment:
> histogram_quantile(.9, rate(histo[5m])) yields the 0.9-quantile for everything.

You're doing a implicit sum there, which I think would cause confusion. No other function does that.

I think the confusion here will become a lot more obvious once binary operators support by clauses, as we'll have to deal with (i.e. error out on) clashing labels there.

`a + by (job) b` where `a` and `b` have many timeseries with different instance labels is a mistake.
Having `histogram_quantile by (job)(.9, rate(histo[5m]))` not also be a mistake seems odd to me.

We could always add the implicit sum behaviour later, once we've seen the full extent of the by clauses across the language.